library/common/main_interface.cc

```diff
@@ -88,7 +88,15 @@ envoy_status_t reset_stream(envoy_stream_t stream) {
 }
 
 envoy_status_t set_preferred_network(envoy_network_t network) {
-  Envoy::Network::MobileUtility::setPreferredNetwork(network);
+  envoy_network_t previous = Envoy::Network::Configurator::setPreferredNetwork(network);
+  if (previous != network && previous != ENVOY_NET_GENERIC) {
+    if (auto e = engine()) {
+      e->dispatcher().post([network]() -> void {
+        if (auto e = engine())
+          e->networkConfigurator().refreshDns(network);
```

Comment:
let's add a test with a mock cache to test that a flush happens on network change.

Reply:
That's actually exactly why I made this an object instead of static (and why initially I had it taking the cache in the constructor, but I'm sure we can mock the manager more or less as easily).

I have the start of a test class written.

If this goes green though, I wonder if it might be worth merging so we can start the release for testing over the weekend?

Reply:
Great. Looking forward to the test. I think there is a way to replace singleton's with mocks. I have a vague memory of that. Let me dig it up.


Well, I would like to have the events we are discussing below go out too. So at worst we merge tomorrow? I can review again once you push.

Reply:
@goaway you can use `TestThreadsafeSingletonInjector` to inject a mock manager I believe.